safe-global/safe-wallet-web
# fix: gasEstimation for counterfactual deployments

## What it solves
- We compute the expected native costs wrong by double accounting the `maxPriorityFee`
- Gas estimation for counterfactual transactions sometimes reverts too early for more advanced txs, that require the `msg.sender` to be the undeployed safe proxy

## How this PR fixes it
- fixes `totalFee` calculation by only multiplying `maxFeePerGas * gasLimit`
- Estimates the counterfactual batch together using the `simulate` function of the fallback handler. This way the inner transaction is a correct `delegateCall`

## How to test it
- Create a new counterfactual Safe
- Add an owner as the first transaction
- Enable Recovery as a first transaction
- Observe the expected `gasLimit` costs to be much closer to the actual `gasLimit`

## Screenshots
![Screenshot 2024-02-14 at 16 09 37](https://github.com/safe-global/safe-wallet-web/assets/2670790/44b93fe1-0241-45c0-a0dd-e0b858c8ae79)

## Checklist
